lib/Shell.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Nominatim;
+
+class Shell
+{
+    public function escapeFromArray($aCmd)
+    {
+        $aArgs = array_map(function ($sArg) {
+            if (preg_match('/^-*\w+$/', $sArg)) return $sArg;
+            return escapeshellarg($sArg);
+        }, $aCmd);
+
+        return join(' ', $aArgs);
+    }
```

Comment:
Having a member function here is a bit odd because your class has no state at all.

Either make this a freestanding function. Then you don't need to instantiate \Nominatim\Shell anymore.

Or make this a fully developed execution object with state, that can be used like this:

$oCmd = new \Nominatim\Shell('osm2pgsql');
$oCmd->addParams('-r', 300);
$oCmd->addParams('-d', 'gis', '--somethingelse');
$oCmd->runWithEnv($aEnv);

Both solutions are fine.